src/dagmc/tests/dagmc_unit_tests.cpp

```diff
@@ -0,0 +1,394 @@
+  EXPECT_EQ(return_string, "mat:Steel");
+  return_string = dgm->return_property("mat:Steel/rho:1.8", "rho", ":", false);
+  EXPECT_EQ(return_string, "rho:1.8");
+
+  return_string = dgm->return_property("mat:Steel", "mat", ":", true);
+  EXPECT_EQ(return_string, "Steel");
+  return_string = dgm->return_property("mat:Steel", "rho", ":", true);
+  EXPECT_EQ(return_string, "");
+  return_string = dgm->return_property("mat:Steel/rho:1.8", "mat", ":", true);
+  EXPECT_EQ(return_string, "Steel");
+  return_string = dgm->return_property("mat:Steel/rho:1.8", "rho", ":", true);
+  EXPECT_EQ(return_string, "1.8");
+
+  return_string = dgm->return_property("mat:Steel", "mat");
+  EXPECT_EQ(return_string, "Steel");
+  return_string = dgm->return_property("mat:Steel", "rho");
+  EXPECT_EQ(return_string, "");
+  return_string = dgm->return_property("mat:Steel/rho:1.8", "mat");
+  EXPECT_EQ(return_string, "Steel");
+  return_string = dgm->return_property("mat:Steel/rho:1.8", "rho");
+  EXPECT_EQ(return_string, "1.8");
+}
+
+//---------------------------------------------------------------------------//
+// FIXTURE-BASED TESTS: Tests to make sure that the split_string function
+// behaves as it should
+//---------------------------------------------------------------------------//
+TEST_F(DagmcMetadataTest, TestSplitString) {
+  // new metadata instance
+  dgm = new dagmcMetaData(DAG);
+
+  std::string to_split = "Neutron/1.0";
+  std::pair<std::string, std::string> pair = dgm->split_string(to_split, "/");
+  EXPECT_EQ(pair.first, "Neutron");
+  EXPECT_EQ(pair.second, "1.0");
+
+  // more complex example
+  std::string more_complex = "|Neutron/1.0|Photon/2.0|";
+  std::vector<std::string> imps = dgm->unpack_string(more_complex, "|");
+  for (unsigned int i = 0 ; i < 2 ; i++) {
+    std::string split = imps[i];
+    pair = dgm->split_string(split, "/");
+    if (i == 0) {
+      EXPECT_EQ(pair.first, "Neutron");
+      EXPECT_EQ(pair.second, "1.0");
+    } else if (i == 1) {
+      EXPECT_EQ(pair.first, "Photon");
+      EXPECT_EQ(pair.second, "2.0");
+    }
+  }
+}
+
+// test to make sure the function try_to_make_int works
+TEST_F(DagmcMetadataTest, TestTryToMakeInt) {
+  // new metadata instance
+  dgm = new dagmcMetaData(DAG);
+
+  EXPECT_EQ(dgm->try_to_make_int("1"), true);
+  EXPECT_EQ(dgm->try_to_make_int("1A"), false);
+  EXPECT_EQ(dgm->try_to_make_int("M33"), false);
+
+}
+// assert some behaviors
+
+class DagmcMetadataTestImplCompMat : public ::testing::Test {
+ protected:
+
+  // initalize variables for each test
+  virtual void SetUp() {
+    // Default h5m file for testing
+    std::string infile = "test_dagmc_impl.h5m";
+
+    DAG = new moab::DagMC();
+
+    rloadval = DAG->load_file(infile.c_str());
+    assert(rloadval == moab::MB_SUCCESS);
+
+    // DAG call to initialize geometry
+    rval = DAG->init_OBBTree();
+    assert(rval == moab::MB_SUCCESS);
+  }
+
+  virtual void TearDown() {
+    //    delete dgm;
```

Comment:
Potential Memory Leak: It looks like `dgm` is allocated (`new`'ed) for each test.  It might be best to re-enable this code and delete it here.

If `dgm` is managed another way, it would be best to delete this commented-out code.